docs/en/setup/service-agent/java-agent/Supported-list.md

```diff
@@ -31,6 +31,7 @@ metrics based on the tracing data.
   * JRE HttpURLConnection (Optional²)
 * HTTP Gateway
   * [Spring Cloud Gateway](https://spring.io/projects/spring-cloud-gateway) 2.0.2.RELEASE -> 3.x (Optional²)
+  * [Apache ShenYu](https://shenyu.apache.org) 2.4.x
```

Comment:
Consider move this whole plugin into optional? Like Apring Gateway.

And update optional-plugins doc ref to Spring gateway.

> Plugin of Spring Cloud Gateway 2.1.x in optional plugin folder. Please only active this plugin when you install agent in Spring Gateway. spring-cloud-gateway-2.x-plugin and spring-webflux-5.x-plugin are both required.

This should be more clear.
If you want the agent out of out for your users, consider to distribute our agent in your binary tar, Seata and DolphinScheduler did this, and activate your required plugin only.

Reply:
Notice `ServerWebExchangeInstrumentation` in gateway-3.x plugin, as well as its shade in pom.xml. It shades the webflux plugin and add a witness class.

Reply:
good idea. It has been remove to optional plugin folder. thinks.